lib/utils.pm

```diff
@@ -1078,7 +1097,22 @@ sub set_hostname {
     assert_script_run "uname -n|grep $hostname";
     systemctl 'status network.service';
     save_screenshot;
-    assert_script_run "if systemctl -q is-active network.service; then systemctl reload-or-restart network.service; fi";
+
+    if (systemctl('is-active NetworkManager', ignore_failure => 1) == 0) {
+        assert_script_run 'nmcli -w 30 network off';
+        assert_script_run 'nmcli -w 30 network on';
+
+        # Despite -w above, NM does not wait for the DHCP lease
+        if (get_var('USE_SUPPORT_SERVER', 0)) {
+            sleep 5;
```

Comment:
You can do `nmcli net conn check` instead. That also waits for connectivity.

Reply:
Thanks that is much better. I have included a loop in case it does not wait for operations in progress. I don't see any indication in the man page it would wait, but it seems to.